util/httputil/http_util.go

```diff
@@ -84,3 +86,58 @@ func AuthTransport(auth *config.Auth, transportConfig *http.Transport) (http.Rou
 
 	return newAuthRoundTripper(auth, transportConfig), nil
 }
+
+func GuessKialiURL(r *http.Request) string {
+	cfg := config.Get()
+
+	// Take default values from configuration
+	schema := cfg.Server.WebSchema
```

Comment:
```suggestion
	schema := "http"
	// if Kiali has been given an identity, then we know it will be using https (since it is via https that its identity gets known)
	if cfg.Identify.CertFile != "" && cfg.Identity.PrivateKeyFile != "" {
		schema = "https"
	}
```

Reply:
For explanation, see comment: https://github.com/kiali/kiali-operator/pull/42#discussion_r433946709

I think the identity setting configures the protocol that Kiali uses to listen, but that is not necessarily the public-facing protocol.